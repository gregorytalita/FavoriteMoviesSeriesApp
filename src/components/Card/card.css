.card {
  width: 250px;
  height: 350px;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  align-items: center;
  margin: 32px 16px;
  position: relative;
  overflow: hidden;
}

.card:hover > .cardContent {
  bottom: 0
}

.cardOverlay {
  width: 100%;
  height: 100%;
  position: absolute;
  background: radial-gradient(transparent, #000000ab);
}

.cardRate {
  z-index: 2;
  display: flex;
  justify-content: space-around;
  width: 100%;
  padding: 16px;
}

.cardContent {
  position: absolute;
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  background: #FFFFFF;
  padding: 32px 24px;
  bottom: -450px;
  transition: all .8s ease-in-out;
  border-top: 8px solid #ec6b74;
  z-index: 3;
}

.cardInfo {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  height: 150px;
  padding: 32px 24px;
}
