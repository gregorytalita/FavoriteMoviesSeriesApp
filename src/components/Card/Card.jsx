import React from 'react'
import { ReactComponent as Remove } from '../../assets/remove.svg'
import { ReactComponent as Tv } from '../../assets/tv.svg'
import { ReactComponent as Movie } from '../../assets/movie.svg'
import { ReactComponent as Rate } from '../../assets/rate.svg'
import {
  makeStyles,
  Paper,
  Button,
  Typography
} from '@material-ui/core'
import './card.css'

const useStyles = makeStyles(() => ({
  button: {
    width: 142,
    height: 56,
    display: 'flex',
    justifyContent: 'space-around',
    alignItems: 'center'
  }
}))

const Card = (props) => {
  const classes = useStyles()

  return (
    <Paper
      className='card'
      elevation={3}
      style={{ background: `url(https://image.tmdb.org/t/p/w300/${props.poster_path})` }}
    >
      <div className='cardOverlay' />

      <div className='cardRate'>
        <Typography style={{ color: '#fff' }}>
          <Rate /> {props.vote_average}
        </Typography>
        {
          props.media_type === 'tv'
            ? <Tv />
            : <Movie />
        }
      </div>

      <div className='cardContent'>
        <div className='cardInfo'>
          <Typography align='center'>
            <b>{props.title}</b>
          </Typography>
          <Typography>
            {props.release_date}
          </Typography>
        </div>
        <Button color='secondary' onClick={props.remove} className={classes.button}>
          <Remove /> Remove
        </Button>
      </div>

    </Paper>
  )
}

export default Card
