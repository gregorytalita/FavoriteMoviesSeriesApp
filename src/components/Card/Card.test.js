import React from 'react'
import { create } from 'react-test-renderer'
import Card from './Card'

const MovieData = {
  media_type: 'movie',
  poster_path: '/AtsgWhDnHTq68L0lLsUrCnM7TjG.jpg',
  release_date: '2019-03-06',
  title: 'Captain Marvel',
  user: 'talitagregory@hotmail.com',
  vote_average: 7
}

describe('Card component', () => {
  test('it matches the snapshot', () => {
    const component = create(<Card {...MovieData} />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
