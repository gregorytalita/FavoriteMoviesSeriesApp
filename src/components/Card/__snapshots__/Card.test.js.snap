// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component it matches the snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation3 card MuiPaper-rounded"
  style={
    Object {
      "background": "url(https://image.tmdb.org/t/p/w300//AtsgWhDnHTq68L0lLsUrCnM7TjG.jpg)",
    }
  }
>
  <div
    className="cardOverlay"
  />
  <div
    className="cardRate"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
      style={
        Object {
          "color": "#fff",
        }
      }
    >
      <svg>
        rate.svg
      </svg>
       
      7
    </p>
    <svg>
      movie.svg
    </svg>
  </div>
  <div
    className="cardContent"
  >
    <div
      className="cardInfo"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
      >
        <b>
          Captain Marvel
        </b>
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        2019-03-06
      </p>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root makeStyles-button-1 MuiButton-text MuiButton-textSecondary"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg>
          remove.svg
        </svg>
         Remove
      </span>
    </button>
  </div>
</div>
`;
